featurebyte/session/session_helper.py

```diff
@@ -88,16 +92,11 @@ async def run_coroutines(coroutines: List[Coroutine[Any, Any, Any]]) -> List[Any
     Exception
         If any task failed
     """
-    tasks = [asyncio.create_task(coro) for coro in coroutines]
+    future = asyncio_gather(*coroutines, max_concurrency=MAX_QUERY_CONCURRENCY)
     try:
-        return cast(List[Any], await asyncio.gather(*tasks))
+        return await future
     except Exception:
-        logger.error(
-            "Canceling all other tasks because at least one task failed",
-            exc_info=True,
-        )
-        for task in tasks:
-            task.cancel()
+        future.cancel()
```

Comment:
The tasks are already cancelled on error in `asyncio_gather`. Is this cancel still required?

Reply:
These tasks are the ones created for the coroutines returned by the wrapper function `_coro` and are different from the tasks in `asyncio_gather` which are created by `asyncio.create_task(coro)`

Reply:
I think you are right this is not necessary since the wrapper tasks should all terminate when the task they created are canceled. Will remove this